jenkins/spark-premerge-build.sh

```diff
@@ -37,27 +37,42 @@ mvn_verify() {
     # file size check for pull request. The size of a committed file should be less than 1.5MiB
     pre-commit run check-added-large-files --from-ref $BASE_REF --to-ref HEAD
 
+    # Get Spark shim versions
+    . $(dirname "$0")/common.sh
+    # snapshots + noSnapshots
+    get_spark_shim_versions -Psnapshots
+    SPARK_SHIM_VERSIONS_ALL=("${SPARK_SHIM_VERSIONS[@]}")
+    # noSnapshots only
+    get_spark_shim_versions -PnoSnapshots
+    SPARK_SHIM_VERSIONS_NOSNAPSHOTS=("${SPARK_SHIM_VERSIONS[@]}")
+
+    # build and run unit tests on one 3.1.X version (base version covers this), one 3.2.X and one 3.3.X version
+    SPARK_SHIM_VERSIONS_TEST=($(for version in ${SPARK_SHIM_VERSIONS_ALL[@]}; do [[ $version != 31* ]] && echo $version; done | sort -n -k1.1,1.2 -u -s))
```

Comment:
This is still some hardcoding that will make us update this line when the base version moves past 3.1.x to say 3.2.0. I think you can already use SPARK_BASE_SHIM_VERSION from version-def.sh instead of "31*" 
or you can use the tail slice  of the array `${SPARK_SHIM_VERSIONS_ALL[@]:1}`  without a filter